File: Data Structures & Algorithms/permutation-string/submission-0.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
      int n= s2.size();
      int k= s1.size();
         sort(s1.begin(), s1.end());
         int i=0; 
         string temp="";
         for(int j=0; j<n; j++){
              if(j-i+1 >k){
                     i++;
              }
           string temp=    s2.substr(i, j-i+1);
           sort(temp.begin(), temp.end());

           if(temp== s1)return true;
         }

         return false;
    }
};

File: Data Structures & Algorithms/copy-linked-list-with-random-pointer/submission-0.java
/*
// Definition for a Node.
class Node {
    int val;
    Node next;
    Node random;

    public Node(int val) {
        this.val = val;
        this.next = null;
        this.random = null;
    }
}
*/

class Solution {
    public Node copyRandomList(Node head) {
            Node newH= null;
            Node curr= head;
          Map<Node, Node>mp= new HashMap<>();
            Node prev= null;

            while(curr!= null){
                 Node temp= new Node(curr.val);
                 mp.put(curr, temp);
                 if(newH==null){
                     newH= temp;
                     prev= newH;
                 }
                 else{
                    prev.next= temp;
                    prev= temp;
                 }
                    curr= curr.next;
            }

          
            Node temp= newH;
            curr = head;
            while(newH!= null){
                newH.random= mp.get(curr.random);
                newH= newH.next;
                curr= curr.next;
            }
           return temp;
    } 
}
